app/models/user.js

```diff
@@ -195,7 +157,14 @@ exports.addModel = function(database) {
       that.id = uuid.v4()
 
       that.validateOnCreate()
-        .then(function(user) { return user.updateHashedPassword() })
+        .then(function(user) {
+          if (_.isFunction(user.hashedPassword)) {
+            var pwd = user.hashedPassword()
+            return user.updatePassword(pwd, pwd)
+          } else {
+            resolve(user)
```

Comment:
so, this one seems to not problematic at all.

`Promise.resolve` is an alias to `Promice.cast`, which is a [static method](https://github.com/petkaantonov/bluebird/blob/master/src/promise.js#L172)